src/shell.cpp

```diff
@@ -0,0 +1,58 @@
+#include "shell.h"
+
+void Shell::runCommand(const std::string& command, bool inBackground) {
+    std::vector<std::string> args = parser.split(command, ' ');
+    std::vector<char*> cargs;
+    for (const auto& arg : args) {
```

Comment:
This logic that converts args to char* looks like it could be in a separate function